File: JavaScript/comp.js
/*
|--------------------------------------------------------------------------
| Are they the "same"?
|--------------------------------------------------------------------------
|
*/

function comp(array1, array2) {
   if (array1 == null || array2 == null) {
      return false;
   } else {
      const a1 = array1.map(i => i ** 2).sort((a, b) => a - b);
      const a2 = array2.sort((a, b) => a - b);
      return JSON.stringify(a1) === JSON.stringify(a2);
   }
}


function comp(a, b) {
   return !!a && !!b && a.map(x => x * x).sort().join() == b.sort().join();
}


function comp(array1, array2) {
   let arr1 = array1 ? array1.map(e => e * e).sort((a, b) => a - b) : 1;
   let arr2 = array2 ? array2.sort((a, b) => a - b).join('') : 1;
   return arr1.join('') == arr2;
}


function comp(a, b) {
   if (!a || !b || a.length !== b.length) return false;
   return a.map(x => x * x).sort().toString() === b.sort().toString();
}


function comp(a1, a2) {
   return a2 != null && a1.map(x => x * x).sort().join('-') == a2.sort().join('-')
}


a1 = [121, 144, 19, 161, 19, 144, 19, 11];
a2 = [11 * 11, 121 * 121, 144 * 144, 19 * 19, 161 * 161, 19 * 19, 144 * 144, 19 * 19];
// a1 = [2, 3, 2]
// a2 = [4, 9, 6]
console.log(comp(a1, a2), true);

File: Python/6py 1/comp.py
"""
|--------------------------------------------------------------------------
| Are they the "same"?
|--------------------------------------------------------------------------
|
"""

def comp(array1, array2):
	if (array1 == None or array2 == None):
		return False
	else:
		a1 = sorted(map(lambda i: i ** 2, array1))
		a2 = sorted(array2)
		return a1 == a2


def comp(array1, array2):
	try:
		return sorted([i ** 2 for i in array1]) == sorted(array2)
	except:
		return False


def comp(a1, a2):
	return None not in (a1,a2) and [i*i for i in sorted(a1)]==sorted(a2)


def comp(array1, array2):
	if array1 and array2:
		return sorted([x*x for x in array1]) == sorted(array2)
	return array1 == array2 == []


def comp(xs, ys):
	if xs is None or ys is None:
		return False
	return sorted(x * x for x in xs) == sorted(ys)


a1 = [121, 144, 19, 161, 19, 144, 19, 11]
a2 = [11*11, 121*121, 144*144, 19*19, 161*161, 19*19, 144*144, 19*19]
print(comp(a1, a2), True)